DiagnosticsExtension/Models/ConnectionStringValidator/MySqlValidator.cs

```diff
@@ -0,0 +1,91 @@
+﻿using DiagnosticsExtension.Controllers;
+using DiagnosticsExtension.Models.ConnectionStringValidator.Exceptions;
+using MySql.Data.MySqlClient;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace DiagnosticsExtension.Models.ConnectionStringValidator
+{
+    public class MySqlValidator : IConnectionStringValidator
+    {
+        public string ProviderName => "MySql.Data.MySqlClient";
+
+        public ConnectionStringType Type => ConnectionStringType.MySql;
+
+        public bool IsValid(string connStr)
+        {
+            try
+            {
+                var builder = new MySqlConnectionStringBuilder(connStr);
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        async public Task<ConnectionStringValidationResult> Validate(string connStr, string clientId = null)
+        {
+            var response = new ConnectionStringValidationResult(Type);
+
+            try
+            {
+                try
+                {
+                    var builder = new MySqlConnectionStringBuilder(connStr);
+                }
+                catch (Exception e)
+                {
+                    throw new MalformedConnectionStringException(e.Message, e);
+                }
+
+                var result = await DatabaseTestController.TestMySqlConnectionString(connStr, null, clientId);
```

Comment:
>DatabaseTestController [](http://example.com/codeflow?start=35&length=22)

Calling another MVC controller in this class looks a bit wierd and is a bit of an anti pattern. 

I suggest moving the TestMySqlConnectionString method to either MySqlVaditor.cs and then have the DatabaseController call this function or put them in a separte class and invoke the methods from databasetestcontroller and this class.

Same with TestSqlServerConnectionString method.

Reply:
done